### DISCUSSION TITLE
How to with sshkey forward based user>

### DISCUSSION THREAD

**Discussion by joeldq** [axivo/k3s-cluster#283] [Discussions]
### Discussion Topic

Configuration

### Description

How to do cluster_vars.ssh.authorized_key  when the running user has SSH access key to all nodes by the key came from ssh key forward (thus no local private key file)

### Code of Conduct

- [X] I agree to follow project's Code of Conduct

**Comment by fmunteanu** [comment]
Basically, the deployment is done from your local computer, as long the SSH key path is defined, it will work. You can hardcode the value, I'm using a Mac, so I prefix the `HOME` variable. Where is your SSH key located? If the key is remote, you can simply copy it locally, on the Ansible execution server.

Also please note that there was recent variables restructure, to simplify the usage, in your case:

https://github.com/axivo/k3s-cluster/blob/91f6590527c6de3b4ed3e5519d679ea197ef0a1b/roles/cluster/defaults/main.yaml#L34

New logic implemented:

- `defaults/main.yaml` contains the user friendly variables
- `tasks/facts.yaml contains` the user advanced variables and other globalized role definitions
